packages/nativewind/src/metro/typescript.ts

```diff
@@ -0,0 +1,67 @@
+import { parse, CommentArray, stringify } from "comment-json";
+import { existsSync, readFileSync, writeFileSync } from "fs";
+import { bold, cyan } from "./picocolors";
+
+export function setupTypeScript(envPath: string) {
+  try {
+    const ts = require("typescript");
+    if (!ts) {
+      return;
+    }
+
+    const configFileName = ts.findConfigFile(
+      "./",
+      ts.sys.fileExists,
+      "tsconfig.json",
+    );
+
+    const userConfig = parse(
+      readFileSync(configFileName, {
+        encoding: "utf-8",
+      }),
+    );
+
+    if (
+      typeof userConfig !== "object" ||
+      !userConfig ||
+      Array.isArray(userConfig)
+    ) {
+      return;
+    }
+
+    const output: string[] = [];
+    let updatedConfig = false;
+
+    if (!existsSync(envPath)) {
+      writeFileSync(
+        envPath,
+        `/// <reference types="nativewind/types" />
+
+// NOTE: This file should not be edited and should be committed with your source code. It is generated by NativeWind.`,
+      );
+      output.push(`Created ${cyan(envPath)}`);
+    }
+
+    userConfig.include ??= new CommentArray(envPath);
+    if (
+      Array.isArray(userConfig.include) &&
+      !userConfig.include.includes(envPath)
+    ) {
+      userConfig.include.push(envPath);
+      updatedConfig = true;
+      output.push(
+        `Updated ${configFileName} to include the ${cyan(envPath)} file`,
+      );
+    }
+
+    if (updatedConfig) {
+      writeFileSync(configFileName, stringify(userConfig, null, 2));
+    }
+
+    if (output.length) {
+      console.log(
+        `${cyan(bold("NativeWind"))} made the following changes to your project to support TypeScript:\n  - ${output.join("\n  - ")}`,
+      );
+    }
+  } catch {}
```

Comment:
@marklawlor maybe worth providing a catch case in the event something does go wrong so the user knows that the configuration didn't properly complete? If for no other reason than to help us debug?
```suggestion
  } catch {
    console.log(Something went wrong while configuring your project to support Typescript. If you create an issue, please provide a screenshot of these logs.)
  }
```

Reply:
might consider using a red color for the error text?

Reply:
There are many reasons why this `try {} catch {}` will fail - including this not being a TypeScript project!

I consider the manual setup the primary way to setup TypeScript. If this works, then thats just a bonus.

I don't really want to deal with issues about this because then you open the door to weird TS issues (e.g using `tsconfig.dev.json` 🙄 )